### PR TITLE
Clean up reference section

### PR DESCRIPTION
**Pull request by raghvenders**
Fix and changes with respect to the issue - Tidy reference section
#22385

Proposed solution mentioned there

- make sure that every item in the left-hand navigation is referenced in the main text of the reference section (not counting the autogenerated list)
 **For this , I tried restructure based on my own preference and combined what is mentioned I already did a PR with quite hundreds of files changes which is recommended to be split and to address for other locals.Kindly help me refer a PR for such kind i can have a look and try or I can take a screenshot of my proposed changes (UI pages and content links) and I will attach it here.**

- set no_list: true in the front matter of content/en/docs/reference/_index.md
  **(Done)**

### CODE REVIEW

**Review by sftim — CHANGES_REQUESTED** [review]
> make sure that every item in the left-hand navigation is referenced in the main text of the reference section 

@raghvenders this PR doesn't yet achieve that aim. To make that change, you would need to edit the text of `content/en/docs/reference/_index.md` to include any links that are in the autogenerated list but not in the preview page.

To see a list of the differences, look at https://deploy-preview-24727--kubernetes-io-master-staging.netlify.app/docs/reference/ and compare it to https://k8s.io/docs/reference/

For example, the revised page does not have a link to `/docs/reference/access-authn-authz/` in the main part of the page.

**Comment by raghvenders** [comment]
@sftim  - Sure tim.I would like to show you sketch of it (Screeshot) with re-arrange hierarchical order or combine them wherever and modify the main section accordingly.

Once you are fine with those , i will apply those changes

**Comment by raghvenders** [comment]
Please find the screenshot with the name changes and grouping in the left [navigation.]
Reference_Left_Navigation_links


Accordingly , I will refer them in the main section (I will update and show you screenhot for main/Overview section as well).

Please let me know if you are fine with first navigation and grouping.

**Comment by raghvenders** [comment]
@sftim  - Any thoughts above re-grouping in reference section.

**Comment by sftim** [comment]
Hi @raghvenders 

I wasn't able to imagine the changes you wanted to describe.

So long as the final PR includes links that are in the autogenerated list, that's probably OK. If necessary we can tweak what goes into what group in a subsequent pull request.

**Comment by raghvenders** [comment]
@sftim  - have done the updates just comparing reference link and main overview section.
I am not sure why PROW didn't build with the updates yet.

**Comment by sftim** [comment]
BTW the tool that builds the website is “Netlify”, not “Prow”. Prow does have other roles though.

My review from https://github.com/kubernetes/website/pull/24727#pullrequestreview-517194534 stands @raghvenders - _every_ page that's in the autogenerated list must survive this edit.

**Comment by raghvenders** [comment]
Thanks @sftim

**Comment by raghvenders** [comment]
thanks @tengqm

**Comment by sftim** [comment]
I'm worried that these changes represent a regression, because we're removing some links. Readers on a mobile device won't know what they're missing.

@raghvenders if it helps I can provide a line-by-line review of the existing set of links and whether they're retained. If you can do that check yourself, that's even better.

**Comment by raghvenders** [comment]
@sftim  - can you please be specific on link readers in mobile device.

Actually we are not removing any specific link rather moving tools.md - > tools/_index.md like every other links under reference section , otherwise correct me if i am missing anything here.

**Comment by sftim** [comment]
I haven't had time yet to list what needs fixing. Sorry about that.

**Comment by kbhawkey** [comment]
Page previews:
https://deploy-preview-24727--kubernetes-io-master-staging.netlify.app/docs/home/
https://deploy-preview-24727--kubernetes-io-master-staging.netlify.app/docs/reference/
https://deploy-preview-24727--kubernetes-io-master-staging.netlify.app/docs/reference/using-api/

File: content/en/docs/reference/tools/_index.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
 ---
+title: Tools
 reviewers:
 - janetkuo
-title: Tools
 content_type: concept
+weight: 80
+no_list: true
+card:
+  name: reference
```
Comment (kbhawkey) [inline content/en/docs/reference/tools/_index.md:9]: Hi @raghvenders . 

I don't know if you need the changes in this page?

The `card` field adds this page as a link (`Tools`) to the docs home page. Is this what you intend?

File: content/en/docs/reference/using-api/_index.md [hunk 2]
```diff
@@ -6,10 +6,12 @@ reviewers:
 - jbeda
 content_type: concept
 weight: 10
+no_list: true
 card:
   name: reference
   weight: 50
   title: Overview of API
+
```
Comment (kbhawkey) [inline content/en/docs/reference/using-api/_index.md:14]: nit: You could remove this extra line.

File: content/en/docs/reference/_index.md [hunk 3]
```diff
@@ -19,7 +20,19 @@ This section of the Kubernetes documentation contains references.
 ## API Reference
 
 * [Kubernetes API Reference {{< latest-version >}}](/docs/reference/generated/kubernetes-api/{{< latest-version >}}/)
-* [Using The Kubernetes API](/docs/reference/using-api/) - overview of the API for Kubernetes.
+* [Kubernetes API Overview](/docs/reference/using-api/) - overview of the API for Kubernetes.
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:23]: @raghvenders , Are these the section to add (keep the text to match the page titles):

Standardized Glossary
Kubernetes Issues and Security
API Access Control
Scheduling
Tools (single page)

Currently, `Scheduling` is listed under the Component Reference section on this page as:

```
 * [kube-scheduler Policies](/docs/reference/scheduling/policies)
  * [kube-scheduler Profiles](/docs/reference/scheduling/config#profiles)
```
Perhaps these links could be pulled out and put under a `Scheduling` heading?

**Comment by sftim** [comment]
I recommend more cleanup. @raghvenders do you have any question(s) following @kbhawkey's most recent comments?

/lgtm cancel

**Comment by raghvenders** [comment]
@sftim - Sure , I will respond to that Tim.

File: content/en/docs/reference/tools/_index.md (hunk 1 above)
Comment (raghvenders) [reply content/en/docs/reference/tools/_index.md:9]: The intention is to keep Tools as not selected link like other links but it appears otherwise.
You can verify it as in the left navigator Reference listing. , so I created and moved it under Tools Folder. 

The intention is not to move it to the Home page but to feel encouraged if it is agreeable and helps.

File: content/en/docs/reference/_index.md (hunk 3 above)
Comment (raghvenders) [reply content/en/docs/reference/_index.md:23]: This makes sense, I will add the changes.

Comment (raghvenders) [reply content/en/docs/reference/_index.md:24]: @kbhawkey  - addressed this review comments

**Comment by raghvenders** [comment]
@kbhawkey @sftim  - changes are done. Please do the review and let me know. I will rebase with the master.

File: content/en/docs/reference/tools/_index.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/reference/tools/_index.md:9]: Hi @raghvenders . Thanks for the updates. Here is some feedback:
- I see the new links added to the reference _index page. The order is slightly different from the left hand navigation tree.
- There is no longer a link to the `Tools` page. Do you want to keep a link to the Tools page on this _index page or leave this link off the page?
- Do you want to add the `Tools` link to the Docs home page?

You could remove the card data or leave the data to add the new `Tools` link,

```
card:
  name: reference
```

Comment (raghvenders) [reply content/en/docs/reference/tools/_index.md:9]: Sure @kbhawkey, I will pull the master to have links update on top.

File: content/en/docs/reference/_index.md [hunk 4]
```diff
@@ -19,7 +21,19 @@ This section of the Kubernetes documentation contains references.
...
+
+## API Access Control
+
+* [API Access Control](/docs/reference/access-authn-authz/) - Overview of how Kubernetes implements and Controls API Access
+
+## Kubernetes Issues and Security
+* [Kubernetes Issue Tracker](/docs/reference/issues-security/issues/)
+* [Kubernetes Security and Disclosure Information](/docs/reference/issues-security/security/)
 
 ## API Client Libraries
```
Comment (sftim) [inline content/en/docs/reference/_index.md:38]: There are 6 official libraries (the existing page is out of date); see https://k8s.io/docs/reference/using-api/client-libraries/#officially-supported-kubernetes-client-libraries

Maybe change “Officially supported client libraries” to a hyperlink, targeting https://k8s.io/docs/reference/using-api/client-libraries/#officially-supported-kubernetes-client-libraries ?

File: content/en/docs/reference/_index.md [hunk 5]
```diff
@@ -19,7 +21,19 @@ This section of the Kubernetes documentation contains references.
 ## API Reference
 
 * [Kubernetes API Reference {{< latest-version >}}](/docs/reference/generated/kubernetes-api/{{< latest-version >}}/)
-* [Using The Kubernetes API](/docs/reference/using-api/) - overview of the API for Kubernetes.
+* [Kubernetes API Overview](/docs/reference/using-api/) - overview of the API for Kubernetes.
+
+## Standardized Glossary
+
+* [Standardized Glossary](/docs/reference/glossary/) - This glossary is intended to be a comprehensive, standardized list of Kubernetes terminology
```
Comment (sftim) [inline content/en/docs/reference/_index.md:28]:
```suggestion
* [Glossary](/docs/reference/glossary/) -  a comprehensive, standardized list of Kubernetes terminology
```

File: content/en/docs/reference/_index.md [hunk 6]
```diff
@@ -19,7 +21,19 @@ This section of the Kubernetes documentation contains references.
...
+
+## API Access Control
+
+* [API Access Control](/docs/reference/access-authn-authz/) - Overview of how Kubernetes implements and Controls API Access
```
Comment (sftim) [inline content/en/docs/reference/_index.md:32]: * [API Access Control](/docs/reference/access-authn-authz/) - details on how Kubernetes controls API access

**Review by sftim — COMMENTED** [review]
Overall, I'm not convinced that we need to keep the level 2 headings in this page. How about removing them?

File: content/en/docs/reference/_index.md [hunk 7]
```diff
@@ -19,7 +21,19 @@ This section of the Kubernetes documentation contains references.
 ## API Reference
 
 * [Kubernetes API Reference {{< latest-version >}}](/docs/reference/generated/kubernetes-api/{{< latest-version >}}/)
-* [Using The Kubernetes API](/docs/reference/using-api/) - overview of the API for Kubernetes.
+* [Kubernetes API Overview](/docs/reference/using-api/) - overview of the API for Kubernetes.
```
Comment (sftim) [inline content/en/docs/reference/_index.md:24]: Could do this:
```suggestion
## Kubernetes API
* [API overview](/docs/reference/using-api/)
* [Kubernetes API reference {{< latest-version >}}](/docs/reference/generated/kubernetes-api/{{< latest-version >}}/)
* [API access control](/docs/reference/access-authn-authz/) - details on how Kubernetes controls API access
* [Well-Known Labels, Annotations and Taints](/docs/reference/kubernetes-api/labels-annotations-taints/)
```

(possibly omitting the heading line)
If you make this change @raghvenders, remove the “API Access Control” section.

**Comment by raghvenders** [comment]
@sftim  - I will work on those review comments. I did a re-base yesterday, but I see PROW needs a re-base.

what is "Superseded"?

**Comment by sftim** [comment]
> Superseded

means the review is no longer relevant because changes since the review make it out of date

File: content/en/docs/reference/_index.md (hunk 7 above)
Comment (raghvenders) [reply content/en/docs/reference/_index.md:24]: @sftim  - I like this by doing this is it fine to make the left navigator link go away too for API Reference(API access control & Well-Known Labels, Annotations, and Taints) and remove API Access control from the left navigation section?

**Comment by raghvenders** [comment]
@sftim and @kbhawkey Addressed all the review comments mentioned above - https://github.com/kubernetes/website/pull/24727/commits/0541b52b470fd21a85d6b8678ca5ef54b17c95b7

https://github.com/kubernetes/website/pull/24727#discussion_r541133574

while addressing this in the overview section I had removed API reference and made those "Kubernetes API reference v1.20" and "Well-Known Labels, Annotations and Taints" to do that, I had to remove _index.md.Sorry If i understood this wrong , i do revert.

**Comment by sftim** [comment]
Hi @raghvenders 

Would you have time to rebase this PR against the latest _master_ branch?

**Comment by raghvenders** [comment]
@sftim  - completed rebasing.

**Comment by kbhawkey** [comment]
Hi @raghvenders .
Would you have time to rebase (against the `master` branch)? The `docs/reference/kubernetes-api` directory has been updated recently. I'd like to see your changes move forward. Thanks!

**Comment by kbhawkey** [comment]
Hi @raghvenders .
Are you still working on this PR? Do you need a hand to move forward?

**Comment by raghvenders** [comment]
@kbhawkey - I did a rebase with the website master and merged changes. Let me know if you any issues. Looks like my laptop clock was on Feb-22 for 2b6074e, but git log --oneline see my HEAD as "2b6074e8e (HEAD -> patch-3, origin/patch-3) changes".

File: content/en/docs/reference/_index.md (hunk 4 above)
Comment (kbhawkey) [reply content/en/docs/reference/_index.md:33]: @raghvenders . Thanks for rebasing.
Could you change this heading from "API Client Libraries" to "Officially supported client libraries"?

File: content/en/docs/reference/_index.md [hunk 8]
```diff
@@ -16,11 +18,17 @@ This section of the Kubernetes documentation contains references.
 
 <!-- body -->
 
-## API Reference
+## Standardized Glossary
+
+* [Glossary](/docs/reference/glossary/) -  a comprehensive, standardized list of Kubernetes terminology
+
+
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:25]: These API-related list items (related to "API overview, access, reference") need a heading (not "Standardized Glossary").

File: content/en/docs/reference/_index.md [hunk 9]
```diff
@@ -32,22 +40,28 @@ client libraries:
...
 
-## CLI Reference
+## CLI
 
 * [kubectl](/docs/reference/kubectl/overview/) - Main CLI tool for running commands and managing Kubernetes clusters.
     * [JSONPath](/docs/reference/kubectl/jsonpath/) - Syntax guide for using [JSONPath expressions](https://goessner.net/articles/JsonPath/) with kubectl.
 * [kubeadm](/docs/reference/setup-tools/kubeadm/) - CLI tool to easily provision a secure Kubernetes cluster.
 
-## Components Reference
+## Components
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:52]: What about a link to the Feature gates page under the "Components" heading?

File: content/en/docs/reference/kubectl/_index.md [hunk 10]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "kubectl CLI"
+title: "CLI"
```
Comment (kbhawkey) [inline content/en/docs/reference/kubectl/_index.md:2]: I suggest to revert this change? All pages in this section are related to kubectl. The left navigation lists "CLI" instead of "kubectl CLI"?

**Comment by kbhawkey** [comment]
Thanks @raghvenders . I requested some minor changes to the headings. 🎉

File: content/en/docs/reference/kubectl/_index.md (hunk 10 above)
Comment (sftim) [reply content/en/docs/reference/kubectl/_index.md:2]: “kubectl CLI” feels tautological; how about just “kubectl”?

Comment (kbhawkey) [reply content/en/docs/reference/kubectl/_index.md:2]: Agreed. Was going to suggest that ...

File: content/en/docs/reference/tools/_index.md [hunk 11]
```diff
@@ -1,22 +1,17 @@
 ---
+title: Other Tools
```
Comment (kbhawkey) [inline content/en/docs/reference/tools/_index.md:2]: Not sure why this change is needed.

File: content/en/docs/reference/_index.md [hunk 12]
```diff
@@ -18,11 +20,17 @@ This section of the Kubernetes documentation contains references.
 
 ## API Reference
 
+* [Glossary](/docs/reference/glossary/) -  a comprehensive, standardized list of Kubernetes terminology
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:23]: nit: The Glossary link does not belong under `API Reference`?

**Comment by kbhawkey** [comment]
Thanks @raghvenders . 👍 
I think the changes are fine. If needed, more changes can be added in a follow-up PR.
/lgtm

**Comment by sftim** [comment]
Let's merge this and follow up with an issue to address the remaining concerns
/approve
